### PR TITLE
Correct PSK build error

### PR DESCRIPTION
**Pull request by tmael**
This PR fixes build errors encountered with the following configurations:

```
./configure --enable-enckeys --disable-asn --disable-rsa --disable-ecc --enable-psk

```

### CODE REVIEW

File: wolfssl/wolfcrypt/asn_public.h [hunk 1]
```diff
@@ -187,8 +187,12 @@ enum {
 
     PEM_PASS_READ  = 0,
     PEM_PASS_WRITE = 1,
-};
+#if defined(NO_ASN)
```
Comment (dgarske) [inline wolfssl/wolfcrypt/asn_public.h]: I don't see a reason to add an ifdef check around these. This is the only place they are defined. Let's say NO_ASN is defined and I still want to know `PKCS5_SALT_SZ` or `HMAC_SHA256_OID` for my own purposes? Also having them in the enum list does not increase code size.

File: src/ssl.c [hunk 2]
```diff
@@ -8045,17 +8043,28 @@ WOLFSSL_EVP_PKEY* wolfSSL_d2i_PUBKEY(WOLFSSL_EVP_PKEY** out, unsigned char** in,
         long inSz)
 {
     WOLFSSL_EVP_PKEY* pkey = NULL;
+#if !defined(NO_RSA) ||   \
```
Comment (dgarske) [inline src/ssl.c]: This complex ifdef seems unnecessary. If the issue is "set, but not used" warning then down at the bottom before the "return" add `(void)mem; (void)memSz; (void)out;`. There is another case like this in `wolfSSL_d2i_PrivateKey_EVP`, which should also be fixed.

File: wolfssl/wolfcrypt/asn_public.h (hunk 1 above)
Comment (tmael) [reply wolfssl/wolfcrypt/asn_public.h:190]: @dgarske, 

 error: redeclaration of enumerator ‘PKCS5_SALT_SZ’ and HMAC_SHA256_OID when building without

File: src/ssl.c (hunk 2 above)
Comment (tmael) [reply src/ssl.c:8046]: @dgarske 

The changes are necessary to fix build error with

File: wolfssl/wolfcrypt/asn_public.h (hunk 1 above)
Comment (dgarske) [reply wolfssl/wolfcrypt/asn_public.h:190]: @tmael : I don't see a redeclaration error with that configure. Also those two `PKCS5_SALT_SZ` and `HMAC_SHA256_OID` are only defined in asn_public.h.

Comment (tmael) [reply wolfssl/wolfcrypt/asn_public.h:190]: @dgarske, sorry try again with --disable-asn

Comment (dgarske) [reply wolfssl/wolfcrypt/asn_public.h:190]: @tmael : `--disable-asn` is in the statement above. Can you confirm ./configure and post the errors?

Comment (tmael) [reply wolfssl/wolfcrypt/asn_public.h:190]: @dgarske 
```
  CC       wolfcrypt/src/src_libwolfssl_la-wolfmath.lo
wolfcrypt/src/wc_encrypt.c:243:25: error: unknown type name ‘EncryptedInfo’
 int wc_BufferKeyDecrypt(EncryptedInfo* info, byte* der, word32 derSz,
                         ^~~~~~~~~~~~~
wolfcrypt/src/wc_encrypt.c:306:25: error: unknown type name ‘EncryptedInfo’
 int wc_BufferKeyEncrypt(EncryptedInfo* info, byte* der, word32 derSz,
                         ^~~~~~~~~~~~~
wolfcrypt/src/wc_encrypt.c: In function ‘wc_CryptKey’:
wolfcrypt/src/wc_encrypt.c:429:22: error: ‘HMAC_SHA256_OID’ undeclared (first use in this function); did you mean ‘WC_SHA256’?
                 case HMAC_SHA256_OID:
                      ^~~~~~~~~~~~~~~
                      WC_SHA256
wolfcrypt/src/wc_encrypt.c:429:22: note: each undeclared identifier is reported only once for each function it appears in
Makefile:4494: recipe for target 'wolfcrypt/src/src_libwolfssl_la-wc_encrypt.lo' failed
make[1]: *** [wolfcrypt/src/src_libwolfssl_la-wc_encrypt.lo] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/home/tesfa/wolfssl'
Makefile:3031: recipe for target 'all' failed
make: *** [all] Error 2

```

**Comment by tmael** [comment]
@dgarske, deleted some comments to avoid confusions. Let me know if you need more info on this.

File: wolfssl/wolfcrypt/asn_public.h (hunk 1 above)
Comment (dgarske) [reply wolfssl/wolfcrypt/asn_public.h:190]: @tmael : The issue is `HMAC_SHA256_OID` and `PKCS5_SALT_SZ` are defined in both `asn_public.h` and `asn.h`.

The `PKCS5_SALT_SZ` is only used by wc_encrypt.c, so please move it to wc_encrypt.h.

The `HMAC_SHA256_OID` should only exist in asn.h, but that entire `enum HMAC_Sum` should be moved out of `NO_ASN`.

David

**Comment by tmael** [comment]
Closing since this PR is no longer relevant.

**Pull request closed without merging**